src/main/ngapp/src/app/main/main.component.ts

```diff
@@ -119,21 +127,42 @@ export class MainComponent implements OnInit {
 
   // Function to make a http call to update the user's preferred language
   updateLanguage(selectedLanguage: string) {
-    const localUser: User = {
-      userId: this.currId,
-      name: '',
-      email: '',
-      language: selectedLanguage,
-    };
+    this.languageUpdateLoading = true;
+    const fetchUserPromise = this.chatService.getUser().toPromise();
+    fetchUserPromise.then(user => {
+        let returnedUser = user;
 
-    const putRequest = this.chatService.updateLanguage(localUser).toPromise();
-    putRequest
-      .then((res) => {
-        console.log(res);
-      })
-      .catch((err) => {
+        const localUser: User = {
+            userId: this.currId,
+            name: returnedUser.name,
+            email: returnedUser.email,
+            language: selectedLanguage,
+        };
+
+        const putRequest = this.chatService.updateLanguage(localUser).toPromise();
+        putRequest.then((res) => {
+            this.languageUpdateLoading = false;
+            this.languageUpdateSuccess = true;
+            console.log(res);
+        })
+        .catch((err) => {
+            this.languageUpdateLoading = false;
+            this.languageUpdateError = err;
+            this.languageUpdateFailure = true;
+            console.error(err);
+        });
+    }).catch(err => {
+        this.languageUpdateLoading = false;
```

Comment:
Do you need both success and error?

!this.LanguageupdateLoading && !this.Failure seems to indicate success.